packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -406,27 +398,10 @@ export class JsonLdCredentialFormatService implements CredentialFormatService<Js
     return this.areCredentialsEqual(offerAttachment, requestAttachment)
   }
 
-  public shouldAutoRespondToCredential(
-    agentContext: AgentContext,
-    { credentialAttachment, requestAttachment }: FormatAutoRespondCredentialOptions
-  ) {
-    const credentialAsJson = credentialAttachment.getDataAsJson<W3cVerifiableCredential>()
-    const credential = JsonTransformer.fromJSON(credentialAsJson, W3cVerifiableCredential)
-
-    if (Array.isArray(credential.proof)) {
-      throw new AriesFrameworkError('Credential arrays are not supported')
-    } else {
-      // do checks here
-      try {
-        const requestAsJson = requestAttachment.getDataAsJson<SignCredentialOptionsRFC0593>()
-
-        this.compareCredentialSubject(credential, requestAsJson)
-        this.compareProofs(credential.proof, requestAsJson)
-        return true
-      } catch (error) {
-        return false
-      }
-    }
+  public shouldAutoRespondToCredential() {
+    // The credential is already compared in the processCredential method, for now
+    // we will always return true.
+    return true
```

Comment:
Not the biggest fan of this. I think double verification is better here. I don't think that it is a very expensive operation and if, somehow, this is ever used separate from `processCredential` (via a custom module or something else) it will cause unexpected behaviour.

Reply:
Sure, can add the verification back in. It's just that a lot of the methods are written with the knowledge the knowledge that the processXXX methods are called up front. These handle initial processing and validation of the objects. But I agree it looks weird.

Reply:
Added the check here as well